src/core/announcements/announcementTypes.ts

```diff
@@ -0,0 +1,183 @@
+  dsnpType: DSNPType.Reaction,
+  emoji,
+  fromId,
+  inReplyTo,
+});
+
+/**
+ * DSNPGraphChangeType: an enum representing different types of graph changes
+ */
+export enum DSNPGraphChangeType {
+  Follow = 1,
+  Unfollow = 2,
+}
+
+/**
+ * GraphChangeAnnouncement: an Announcement of type GraphChange
+ */
+export interface GraphChangeAnnouncement extends TypedAnnouncement<DSNPType.GraphChange> {
+  fromId: string;
+  changeType: DSNPGraphChangeType;
+  objectId: string;
+}
+
+/**
+ * createFollowGraphChange() generates a follow graph change announcement from
+ * a given DSNP user id.
+ *
+ * @param   fromId     - The id of the user from whom the announcement is posted
+ * @param   followeeId - The id of the user to follow
+ * @returns A GraphChangeAnnouncement
+ */
+export const createFollowGraphChange = (fromId: string, followeeId: string): GraphChangeAnnouncement => ({
+  fromId,
+  dsnpType: DSNPType.GraphChange,
+  changeType: DSNPGraphChangeType.Follow,
+  objectId: followeeId,
+});
+
+/**
+ * createUnfollowGraphChange() generates an unfollow graph change announcement
+ * from a given DSNP user id.
+ *
+ * @param   fromId     - The id of the user from whom the announcement is posted
+ * @param   followeeId - The id of the user to unfollow
+ * @returns A GraphChangeAnnouncement
+ */
+export const createUnfollowGraphChange = (fromId: string, followeeId: string): GraphChangeAnnouncement => ({
+  fromId,
+  dsnpType: DSNPType.GraphChange,
+  changeType: DSNPGraphChangeType.Unfollow,
+  objectId: followeeId,
+});
+
+/**
+ * ProfileAnnouncement: an Announcement of type Profile
+ */
+export interface ProfileAnnouncement extends TypedAnnouncement<DSNPType.Profile> {
+  contentHash: string;
+  fromId: string;
+  url: string;
+}
+
+/**
+ * createProfile() generates a profile announcement from a given URL and hash.
+ *
+ * @param   fromId - The id of the user from whom the announcement is posted
+ * @param   url    - The URL of the activity content to reference
+ * @param   hash   - The hash of the content at the URL
+ * @returns A ProfileAnnouncement
+ */
+export const createProfile = (fromId: string, url: string, hash: HexString): ProfileAnnouncement => ({
+  dsnpType: DSNPType.Profile,
+  contentHash: hash,
+  fromId,
+  url,
+});
+
+export type AnnouncementWithSignature<T extends Announcement> = T & { signature: string };
+
+export type SignedAnnouncement = AnnouncementWithSignature<Announcement>;
+
+export type SignedBroadcastAnnouncement = AnnouncementWithSignature<BroadcastAnnouncement>;
+export type SignedReplyAnnouncement = AnnouncementWithSignature<ReplyAnnouncement>;
+export type SignedReactionAnnouncement = AnnouncementWithSignature<ReactionAnnouncement>;
+export type SignedProfileAnnouncement = AnnouncementWithSignature<ProfileAnnouncement>;
+export type SignedGraphChangeAnnouncement = AnnouncementWithSignature<GraphChangeAnnouncement>;
```

Comment:
Had the same idea in my current PR. Could you add some doc comments while we're at it?